Created Climesync config file

aaroncohen73 opened the pull request:
Climesync now uses a config file (~/.climesyncrc) to store server url, username, and password information. At the current time, this information is only protected by user permissions (chmod 600).

When the program first starts, arguments passed in through the command line take highest precedence followed by values stored in the configuration file. When users enter new server or user information while using the program, they are prompted by the program to ask if they want to store the new information in the config file.

In addition, the way that the program handles command line arguments in general has been cleaned up and clarified.

Closes #28

matthewrsj commented on climesync.py:
```diff
@@ -34,12 +37,86 @@
+    with open(realpath, "r") as f:
+        for line in f:
+            # Ignore comments (lines beginning with '#')
+            if line.strip()[0] == "#":
```
No comments allowed after first character?

ElijahCaine commented on climesync.py:
```diff
@@ -34,12 +37,86 @@
+    # Keys and values can consist of any character except an equals sign
+    config = re.compile("[^=]+ = [^=]+")
+
+    with open(realpath, "r") as f:
```
Why are we writing regexes to parse our config file instead of just using a library like https://docs.python.org/3.5/library/configparser.html ?

Kennric replied:
Seconded. Please use a standard library for this.

ElijahCaine commented on climesync.py:
```diff
@@ -220,17 +327,39 @@ def disconnect():
+    # If username or password in config, use them at program startup.
+    if arg_user:
+        username = arg_user
+
```
Please remove the spacing between `if/elif/else` blocks so the blocks are easier to parse as being contiguous.
```
if stuff
    foo

else
    baz

if  morestuff
    bar

elif morestuff
    foobee
```
is harder to parse than
```
if stuff
    foo
else
    baz

if  morestuff
    bar
elif morestuff
    foobee
```

aaroncohen73 commented:
@ElijahCaine @matthewrsj I've changed it to use the ConfigParser module now (I didn't know that it existed before). I've also fixed the spacing to make it more readable.

matthewrsj commented on climesync.py:
```diff
@@ -702,31 +767,32 @@ def menu():
-    if args.connect:
-        timesync_url = args.connect
+    # Config file
+    config_dict = {}
```
You don't need this line since you set `config_dict` in both the `try` and `except` blocks below.

matthewrsj commented on climesync.py:
```diff
@@ -170,6 +222,23 @@ def get_fields(fields):
+        return
+
+    print "{} = {}".format(key, value)
+    response = get_field("Add to the config file?", field_type="?")
```
Your default options aren't working:
```
(h for help) $ s
Username: admin
Password: <admin-password>
username = admin
(y/N) Add to the config file?: N
password = <admin-password>
(y/N) Add to the config file?: 
Please submit a valid input
(y/N) Add to the config file?:  
Please submit a valid input
(y/N) Add to the config file?: 
Please submit a valid input
(y/N) Add to the config file?: n

token: ...
```

aaroncohen73 replied:
@matthewrsj What do you mean by that?

matthewrsj replied:
@aaroncohen73 the uppercase `N` here implies that it is the default that is selected if you don't type anything
```
(y/N) Add to the config file?:
```
But I have to type `n` to select a no.

matthewrsj commented:
Just a couple easy-to-fix comments. Other than those this looks great.

aaroncohen73 commented:
@matthewrsj Fixed

matthewrsj commented:
+1. I like that I don't have to manually set a config file and climesync just does it for me.

MaraJade commented:
+1 Looks good to me
